extensions-core/lookups-cached-global/src/main/java/org/apache/druid/server/lookup/namespace/cache/CacheScheduler.java

```diff
@@ -232,7 +243,7 @@ private void tryUpdateCache(String currentVersion) throws Exception
           CacheState previousCacheState = swapCacheState(newVersionedCache);
           if (previousCacheState != NoCache.ENTRY_CLOSED) {
             updatedCacheSuccessfully = true;
-            if (previousCacheState instanceof VersionedCache) {
+            if (previousCacheState instanceof VersionedCache && !(newVersionedCache).incremental) {
```

Comment:
It would be more robust to compare `previousCacheState`'s and `newVersionedCache`'s `cacheHandler`s to be identical. Also, the previous VersionedCache should still "close", it should just be a no-op. I would log a custom message.

Reply:
Done, logged debug message instead of not calling close method.